### ISSUE TITLE
Tensorflow can build and even run a model with `Conv2D('Kernel_size=0' )`

### ISSUE THREAD

**Issue by shiningrain** [tensorflow/tensorflow#37334]
**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Win 10 & Linux Ubuntu18.04
- Tensorflow backend (yes/no): yes
- TensorFlow version:1.15.0(CPU)
- Python version: 3.6.9
- CUDA/cuDNN version: -
- GPU model and memory: -

**Describe the current behavior**  
When I build a model with unreasonable parameters  `Conv2D(kernel_size=0)` on TensorFlow, **it can run normally and even generate/save an model** . When I use this model to predict, Tensorflow spend about 5 minutes and still can't return an output.
`Conv2D(kernel_size=0)`  seems like a corner case because **in the convolution operation, it is impossible to calculate with `kernel_size=0`**

Does `kernel_size=0` have some special meaning in Tensorflow? I have not found any description about this case in documents. If no special meaning, **Should Tensorflow set a check for such unreasonable parameters to avoid the risks and incorrect usages in the model?**  

**Code to reproduce the issue**  

```
import os
import numpy as np
import keras.layers as L
from keras.models import load_model
from keras.engine import Model, Input

kwargs = {'filters': 19, 'kernel_size': 0, 'padding': 'valid', 'strides': (2, 4), 'dilation_rate': 1, 'data_format': 'channels_first'}
input = (10 * np.random.random((1,32,32,16)))
layer = L.convolutional.Conv2D(**kwargs)
x = Input(batch_shape=input.shape)
y = layer(x)
bk_model = Model(x, y)
model_path = os.path.join('./', 'model.h5')
bk_model.save(model_path, bk_model)
model = load_model(model_path)
output = model.predict(input)
print('finish')
```

**Comment by Saduf2019** [comment]
@shiningrain 
I have run the code shared by you and it executes as expected, please find[ gist here](https://colab.sandbox.google.com/gist/Saduf2019/002a441ed92036884b41ae5a9f104210/37334.ipynb)
For kernel_size related information please refer to [this link](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2D#arguments_2), also please refer to this link if it [helps](https://github.com/cwehmeyer/pydpc/issues/2#issuecomment-510912684)

**Comment by shiningrain** [comment]
> @shiningrain
> I have run the code shared by you and it executes as expected, please find[ gist here](https://colab.sandbox.google.com/gist/Saduf2019/002a441ed92036884b41ae5a9f104210/37334.ipynb)
> For kernel_size related information please refer to [this link](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2D#arguments_2), also please refer to this link if it [helps](https://github.com/cwehmeyer/pydpc/issues/2#issuecomment-510912684)

Hi!  @Saduf2019 
Thanks for your help!
However, I still have some problems with your reply.
1. I have read your reproduced code, but the `kernel_size` in it is set to 1, which is not the problem  I want to explain. I have also tested with `kernel_size = 1` and it did work properly. Here I want to focusing on illogical inputs such as `kernel_size = 0.`
2. The description of `kernel_size` in the TensorFlow document you gave just said that **the value can be an integer or an integer tuple**. **The value of 0 does conform to what the document says**, so at least the document is not precise.
3. The [last link](https://github.com/cwehmeyer/pydpc/issues/2) you gave seems to have nothing to do with the convolution operation and does not reflect how TensorFlow handle with the illogical operation such as `kenel_size = 0`.

To sum up, I mean that TensorFlow can still generate models and make predictions with illogical parameters such as `kenel_size = 0`. This should be an implementation issue. Developers should consider this situation to check its value or at least set a default value to avoid such problems.

**Hope to receive your response and thank you very much for your help**

**Comment by Saduf2019** [comment]
please find the [gist](https://colab.sandbox.google.com/gist/Saduf2019/04e40b481d05e6a0c2e36ef9c0887ae6/37334.ipynb) for kernel_size=0

**Comment by ymodak** [comment]
This has been fixed with tf-nightly version.
https://github.com/tensorflow/tensorflow/blob/84eb083bb5328912dde064b8b0f61d28c6edbe43/tensorflow/python/keras/layers/convolutional.py#L132

**Comment by fsx950223** [comment]
The problem still exists if I use nn module or  other tensorflow clients. e.g node.js,Java, CPP

**Comment by ymodak** [comment]
@fsx950223 Can you please create a new issue and provide your repro example? Also refer this issue on the new thread. Thanks!
